src/telemetry.rs

```diff
@@ -13,10 +13,23 @@
 // limitations under the License.
 
 use once_cell::sync::Lazy;
+use once_cell::sync::OnceCell;
 use std::time::Instant;
-use tracing_subscriber::prelude::*;
+use tracing_subscriber::{
+    filter,
+    filter::{EnvFilter, LevelFilter},
+    prelude::*,
+    reload, Layer, Registry,
+};
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("{0}")]
+    RequestFailure(String),
```

Comment:
Can we make an enum entry per error type instead of just shoving a generic String value?